test/unit/test_devices/test_asa_device.py

```diff
@@ -304,6 +306,75 @@ def test_count_teardown(self):
         assert self.count_teardown == 0
 
 
+@mock.patch.object(ASADevice, "redundancy_state", new_callable=mock.PropertyMock)
+@pytest.mark.parametrize(
+    "redundancy_state,expected",
+    (
+        ("active", True),
+        ("standby ready", False),
+        (None, True),
+    ),
+    ids=("active", "standby_ready", "unsupported"),
+)
+def test_is_active(mock_redundancy_state, asa_device, redundancy_state, expected):
+    mock_redundancy_state.return_value = redundancy_state
+    actual = asa_device.is_active()
+    assert actual is expected
+
+
+@pytest.mark.parametrize(
+    "side_effect,expected",
+    (
+        ("show_failover_host_active.txt", STANDBY),
+        ("show_failover_host_standby.txt", ACTIVE),
+        ("show_failover_host_off.txt", "disabled"),
+        ("show_failover_groups_active_active.txt", ACTIVE),
+        ("show_failover_groups_active_standby.txt", STANDBY),
+        ("show_failover_groups_standby_active.txt", ACTIVE),
+        (asa_module.CommandError("show failover", r"% invalid command"), None),
+    ),
+    ids=("standby", ACTIVE, "disabled", "active_active", "active_standby", "standby_active", "none"),
+)
+def test_peer_redundancy_state(side_effect, expected, asa_show):
+    device = asa_show([side_effect])
+    actual = device.peer_redundancy_state
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    "side_effect,expected",
+    (
+        ("show_failover_host_active.txt", "on"),
+        ("show_failover_host_off.txt", "off"),
+        (asa_module.CommandError("show failover", r"% invalid command"), "n/a"),
+    ),
+    ids=("on", "off", "n/a"),
+)
+def test_redundancy_mode(side_effect, expected, asa_show):
+    device = asa_show([side_effect])
+    actual = device.redundancy_mode
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    "side_effect,expected",
+    (
+        ("show_failover_host_active.txt", ACTIVE),
+        ("show_failover_host_standby.txt", STANDBY),
+        ("show_failover_host_off.txt", "disabled"),
+        ("show_failover_groups_active_active.txt", ACTIVE),
+        ("show_failover_groups_active_standby.txt", ACTIVE),
+        ("show_failover_groups_standby_active.txt", STANDBY),
+        (asa_module.CommandError("show failover", r"% invalid command"), None),
+    ),
+    ids=(ACTIVE, "standby", "disabled", "active_active", "active_standby", "standby_active", "none"),
+)
+def test_redundancy_state(side_effect, expected, asa_show):
+    device = asa_show([side_effect])
+    actual = device.redundancy_state
+    assert actual == expected
+
+
```

Comment:
The tests look great and I think they cover all possible combinations of states.